__tests__/routes/note.test.js

```diff
@@ -0,0 +1,371 @@
+const request = require('supertest');
+const express = require('express');
+const db = require('../../data/db-config');
+const authRequired = require('../../api/middleware/authRequired');
+const handleError = require('../../api/middleware/handleError');
+const notesRouter = require('../../api/notes/noteRouter');
+
+beforeAll(async () => {
+  await db.migrate.rollback();
+  await db.migrate.latest();
+});
+beforeEach(async () => await db.seed.run());
+
+afterAll(async () => await db.destroy());
+
+afterEach(() => jest.clearAllMocks());
+
+jest.mock('../../api/middleware/authRequired', () =>
+  jest.fn((req, res, next) => next())
+);
+
+jest.mock('../../api/middleware/permissionsRequired', () => ({
+  adminRequired: jest.fn((req, res, next) => next()),
+  superAdminRequired: jest.fn((req, res, next) => next()),
+}));
+
+const app = express();
```

Comment:
Let's move lines 27-30 towards the top of the file. 

Set up your router before setting up your test suite.

Reply:
line 27-30 moved to line 8.